airsonic-main/src/main/java/org/airsonic/player/service/SecurityService.java

```diff
@@ -55,11 +66,27 @@
 
     @Autowired
     private UserDao userDao;
+
+    @Autowired
+    private SonosLinkDao sonosLinkDao;
+
     @Autowired
     private SettingsService settingsService;
+
     @Autowired
     private Ehcache userCache;
 
+    @Autowired
+    private Ehcache sonosLinkcodeCache;
+
+    @Autowired
+    private JWTSecurityService jwtSecurityService;
+
+    // This is initialize in GlobalSecurityConfig, is autowired.
+    // Something wrong here, some circular ref, maybe rebuild responsibilities...
```

Comment:
Please prefix this comment with `TODO ` if it's something that must get fixed.

Reply:
So I re-evaluate that code, this service must not be get by component scan but declared in global security. And it will be move to the security package. 

All initialisation about security it's set there and the service just use the global component.